File: Python/lark_card_kit/models/elements/select_img.py
"""SelectImg element for image selection."""

from __future__ import annotations

from dataclasses import dataclass, field
from typing import Any

from .base import Element
from .button import ConfirmConfig
from .plain_text import PlainText


@dataclass
class SelectImg(Element):
    """Image selector element for selecting images.

    Allows users to select images from a predefined set.
    """

    tag: str = field(default="select_img", init=False)
    name: str | None = None
    placeholder: PlainText | None = None
    selected_value: str | None = None
    disabled: bool | None = None
    confirm: ConfirmConfig | None = None
    width: str | None = None
    options: list[Any] = field(default_factory=list)

File: TypeScript/LarkCardKit/src/models/elements/SelectImg.ts
import type { ElementBase } from './Element';
import type { PlainText } from './PlainText';

export interface SelectImgOption {
  value: string;
  text?: PlainText;
}

export interface SelectImg extends ElementBase {
  tag: 'select_img';
  name?: string;
  required?: boolean;
  placeholder?: PlainText;
  label?: PlainText;
  select_mode?: 'single' | 'multi';
  initial_options?: string[];
  options?: SelectImgOption[];
  width?: string;
  disabled?: boolean;
  confirm?: ConfirmConfig;
  behaviors?: any[];
}

export interface ConfirmConfig {
  title?: PlainText;
  text?: PlainText;
}
